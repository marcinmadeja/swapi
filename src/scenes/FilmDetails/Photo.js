import React from 'react';
import styled from 'styled-components';
import FilmsImage from 'images/films';

const Img = styled.img`
  width: 100%;
  display: block;
  margin-bottom: 15px;
`;

const Photo = ({ episodeId }) => {
  if (!episodeId) return null;

  return (
    <Img src={FilmsImage[`film${episodeId}`]} alt="" />
  );
};

export default Photo;
