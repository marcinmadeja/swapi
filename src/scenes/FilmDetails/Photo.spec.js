import React from 'react';
import 'jest-styled-components';
import FilmsImage from 'images/films';
import Photo, { Img } from './Photo';

const defaultProps = {
  episodeId: '',
  updateSwVehicles: () => null,
};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const component = <Photo {...props} />;
  const shallowComponent = shallow(component);

  return {
    component,
    shallowComponent,
  };
};


describe('Photo', () => {
  it('should match snapshot', () => {
    const { component } = setup({ episodeId: '1' });
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });

  it('should render null when no episodeId', () => {
    const { shallowComponent } = setup({ episodeId: '' });
    expect(shallowComponent.html()).toEqual(null);
  });
});
