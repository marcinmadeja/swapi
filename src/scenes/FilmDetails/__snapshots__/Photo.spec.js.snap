// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
.c0 {
  width: 100%;
  display: block;
  margin-bottom: 15px;
}

<img
  alt=""
  className="c0"
  src="film-1.jpeg"
/>
`;
